src/tests/unittests/instrument_drivers/test_gates.py

```diff
@@ -66,6 +66,15 @@ def test_get_boundaries(self):
 
         virtual_dac.close()
 
+    def test_restrict_boundaries(self):
+        gate_map = {'P1': (0, 1), 'P2': (0, 2), 'P3': (0, 3)}
+        virtual_dac = VirtualDAC(instrumentName('test'), instruments=[self.ivvi], gate_map=gate_map)
+        gate_boundaries = {'P1': (0, 100), 'P2': (0, 100)}
+        virtual_dac.set_boundaries(gate_boundaries)
+        virtual_dac.restrict_boundaries({'P1': (-50, 50), 'P3': (0, 1)})
```

Comment:
I tried virtual_dac.restrict_boundaries(({'P1': (150, -150)}) here. Do you know what happens then? Is this as expected? In real OO boundary would be a class instead of a dictionary. Within this class relations between the values can be checked on the fly when being set in the constructor for example.

Reply:
Also tried virtual_dac.restrict_boundaries({}) here. There happens more than I expected.

Reply:
I added a check to make sure a boundary (a,b) satisfies a<b.